server/app.py

```diff
@@ -64,18 +65,28 @@ def main():
     tornado.options.define("super_emails", default = "", help = "email of site super administrator (maximum rights)", type = list)
     tornado.options.define("db_path", default = "data.db", help = "path to database file", type = str)
 
-    tornado.options.define("cfg", default = "server.conf", help = "path to configuration file", type = str)
+    tornado.options.define("cfg", default = "", help = "path to configuration file", type = str)
 
+    #parse config
     tornado.options.parse_command_line()
-    if os.path.exists(tornado.options.options.cfg):
-        tornado.options.parse_config_file(tornado.options.options.cfg)
-
+    config = None
+    if tornado.options.options.cfg != "":
+        config = configparser.ConfigParser()
+        config.read(tornado.options.options.cfg)
+    
     # prepare database
     database = db.DataBase()
     database.init()
 
     # preparing for search
-    rocksdb_fm_path = os.path.abspath("../../kb.bin/file_memory")
+    rocksdb_fm_path = os.path.abspath(
+        os.path.join(
+            # path to the config file, excluding filename
+            os.path.dirname(tornado.options.options.cfg),
+            config['Repo']['Path'],
+            "kb.bin/file_memory"
+        )
+    ) if config != None else "../../kb.bin/file_memory"
```

Comment:
Can we avoid relative paths scheme using in code?

Reply:
The reason behind this decision is to provide a meaningful fallback for those people who will try to run the sc-web using outdated guides and instructions - it's important to provide compatibility with older versions of our software AND documentation, though I agree that we should get rid of this hardcode as soon as possible.

The plan was to delete this whole code section after implementing identifier caching over network. Would it be fine with you if I'll add a `TODO:` or\and open an issue, so that we'll fix it after implementing the API for caching to work **properly**? Or you'd rather delete this right now?

Reply:
Very soon, perhaps this code will work differently, when the new file storage is merged. It is necessary to come up with a universal solution - constantly avoiding universal solutions with labels `TODO:` is unacceptable.

Reply:
Well, the universal solution is what we've actually came up with in this PR - a config from which every part of the platform reads the "output path", where "kb.bin" (a standardised folder name for compiled KB) is located and can be read from. 

The code branch you're commenting on is only executed when config file is not provided, after merging the PR it is considered deprecated. I can delete this code, but at the cost of older instructions and installation scripts becoming invalid. If that's the price we're willing to pay - no problem, I'll delete it, and the "universal" solution will remain. I would like to know @KovalM's perspective on this.

P.S A REAL universal solution would be to do this over network, but I am not seeing anyone willing to implement it 🤷🏻‍♂️